suse_migration_services/units/product_setup.py

```diff
@@ -58,29 +58,61 @@ def main():
         products_metadata = os.sep.join(
             [root_path, 'etc', 'products.d']
         )
-        baseproduct = os.sep.join(
-            [products_metadata, 'baseproduct']
-        )
-        if os.path.exists(baseproduct):
+        if os.path.exists(products_metadata):
             log.info('Creating backup of Product data')
             Command.run(
                 [
                     'rsync', '-zav', '--delete', products_metadata + os.sep,
                     '/tmp/products.d.backup/'
                 ]
             )
-            log.info('Updating Base Product to be suitable for migration')
-            xml = ElementTree()
-            xml.parse(baseproduct)
-            register_sections = xml.findall('register')
-            for register in register_sections:
-                target_sections = register.findall('target')
-                for target in target_sections:
-                    register.remove(target)
-            xml.write(
-                baseproduct, encoding="UTF-8", xml_declaration=True
-            )
+        baseproduct_filename = get_baseproduct(products_metadata)
```

Comment:
I think this is a bit misleading. The return from get_baseproduct is not a filename it's a list of filenames. I would change this to return a static tuple like this:

```python
from collections import namedtuple

base_product_type = namedtuple(
    'base_product_type', ['count', 'prod_filename']
)

...

baseproduct = base_product_type(
    count=1, prod_filename="foo"
)

print(baseproduct.count)
print(baseproduct.prod_filename)
```

The advantage is this is more readable and the code checking for the exceptions is more clean.
In case you found multiple baseproducts count is set to a number > 1 and prod_filename is None.
In case you found no baseproduct count is set to 0 and prod_filename is None. In case one baseproduct 
is found count is set to 1 and prod_filename is the name.

You don't have to fiddle with a list here and it makes the code more explicit. Thoughts ?

Reply:
I like the approach. Although, prod_filename will be ignored or irrelevant if count is not 1 as that content is never checked in those cases.